// Copyright 2016 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Sync protocol datatype extension for the reading list items.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// Sync Reading list entry. This proto contains the fields synced for a reading
// list entry. It must be kept synced with the reading_list.ReadingListLocal
// protobuf.
message ReadingListSpecifics {
  optional string entry_id = 1;
  optional string title = 2;
  optional string url = 3;
  optional int64 creation_time_us = 4;
  optional int64 update_time_us = 5;
  optional int64 first_read_time_us = 7;
  optional int64 update_title_time_us = 8;
  optional int32 estimated_read_time_seconds = 9;

  enum ReadingListEntryStatus {
    UNREAD = 0;
    READ = 1;
    UNSEEN = 2;
  }

  // If the field is not present, it defaults to UNSEEN.
  optional ReadingListEntryStatus status = 6;
}
